### PR TITLE
Introduce new Landing Page layout

### PR DESCRIPTION
**Pull request by lmuntaner**
The main motivation is to introduce a new layout for the landing page of II.

In this PR, I introduce the new layouts and prepare the flag to enable it.

Yet, I only enable it in the showcase.

This is because we still need changes in the screens before being ready for release. Those changes will happen in another PR.

<hr/><details><summary>🟢 Some screens were added</summary></details>

### CODE REVIEW

File: src/frontend/src/components/authenticateBox.ts
Comment (lmuntaner) [inline src/frontend/src/components/authenticateBox.ts:569]: I needed this to use it in the showcase for now.

In a following PR, I'll remove it and the `firstTime` and `pick` will always come with the new landing page, whereas the `useExisting` comes with the old layout.

File: src/showcase/src/pages/manageNewLanding.astro
Comment (lmuntaner) [inline src/showcase/src/pages/manageNewLanding.astro:1]: This will be removed in a following PR because the `manage` will already be with the new landing.

But I wanted to have both in the showcase for now to share with product and design as well as myself when developing.

File: src/frontend/src/components/landingPage.ts
Comment (frederikrothenberger) [inline src/frontend/src/components/landingPage.ts:34]: Could you extract the text into an i18n file?

File: src/frontend/src/styles/main.css
Comment (frederikrothenberger) [inline src/frontend/src/styles/main.css:134]: It looks great. But I'm absolutely hopeless with CSS so it might be good to get a second opinion form @peterpeterparker or @meodai.

File: src/frontend/src/components/landingPage.ts
Comment (frederikrothenberger) [inline src/frontend/src/components/landingPage.ts:31]: In the other layout there is a special (hidden) title for accessibility. I think for screen readers the landing page should still say "Internet Identity" first (which it presumably does in the old layout due to the hidden title?).

Or is the accessibility concern appropriately addressed by [this](https://github.com/dfinity/internet-identity/pull/2494#issuecomment-2152207910)?

Comment (lmuntaner) [reply src/frontend/src/components/landingPage.ts:31]: It was addressed by https://github.com/dfinity/internet-identity/pull/2494#issuecomment-2152207910, yes.

The logo appears first with the title Internet Identity.

File: src/frontend/src/components/landingPage.ts
Comment (lmuntaner) [reply src/frontend/src/components/landingPage.ts]: Oh, yes! Thanks!

**Comment by lmuntaner** [comment]
@peterpeterparker @meodai could you take a look at the CSS of the new landing page? Thanks!

File: src/frontend/src/styles/main.css
Comment (lmuntaner) [reply src/frontend/src/styles/main.css:134]: Done!

**Comment by peterpeterparker** [comment]
I don't really know the CSS of II, better if @meodai does have a look.

**Review by frederikrothenberger — APPROVED** [review]
LGTM, thanks! Please wait for feedback from @meodai before merging.

File: src/frontend/src/styles/main.css
Comment (meodai) [reply src/frontend/src/styles/main.css:134]: @lmuntaner: I would use vars here: `linear-gradient(90deg, var(--vc-brand-purple), var(--vc-brand-purple-light));` and the `%` are not needed if you are going to split them evenly.

File: src/frontend/src/styles/main.css
Comment (meodai) [inline src/frontend/src/styles/main.css:207]: --rc is for color, not a gradient. it would be rg or something I guess.

But I think  `--rc-landing-page-background` is too specific anyway. It allows for almost no reuse. Maybe calling it something like --rg-highlight? I would discuss with Artem for what reason the gradients are used and use this as the name,

File: src/frontend/src/styles/main.css
Comment (meodai) [inline src/frontend/src/styles/main.css:261]: reference tokens have to point to value tokens.

File: src/frontend/src/styles/main.css
Comment (meodai) [inline src/frontend/src/styles/main.css:323]: these are also too specific I think. Maybe calling it something like split layout would allow for some reuse?

File: src/frontend/src/styles/main.css
Comment (meodai) [inline src/frontend/src/styles/main.css:973]: should this not be min-height?

Comment (meodai) [reply src/frontend/src/styles/main.css:973]: And dvh instead of vh?

File: src/frontend/src/styles/main.css
Comment (meodai) [inline src/frontend/src/styles/main.css:1012]: should this var not be set more globally or not at all if its only used here?

File: src/frontend/src/styles/main.css
Comment (meodai) [inline src/frontend/src/styles/main.css:1074]: why position static? its the default anyway

File: src/frontend/src/components/landingPage.ts
Comment (meodai) [inline src/frontend/src/components/landingPage.ts:38]: this the main I think not the thing around...

File: src/frontend/src/components/landingPage.ts
Comment (meodai) [inline src/frontend/src/components/landingPage.ts:28]: semantic elements are supposed to have either or an aria-label

**Review by meodai — CHANGES_REQUESTED** [review]

File: src/frontend/src/styles/main.css
Comment (lmuntaner) [reply src/frontend/src/styles/main.css:973]: I didn't know about `dvh` thanks!

I tried with `min-height`. But it creates a weird two scroll with the right page is higher than the screen. With `height` the scroll is smooth.

File: src/frontend/src/styles/main.css
Comment (lmuntaner) [reply src/frontend/src/styles/main.css:1074]: Because I set it to `absolute` for desktop and I want the default for mobile.

File: src/frontend/src/styles/main.css
Comment (lmuntaner) [reply src/frontend/src/styles/main.css:134]: Done! Good point, I didn't realize they were the same colors, thanks!

File: src/frontend/src/styles/main.css
Comment (lmuntaner) [reply src/frontend/src/styles/main.css:207]: I prefer to keep it as `rc`. A color could easily replace the gradient in the future and I don't want to change the name of the variable then.

File: src/frontend/src/styles/main.css
Comment (lmuntaner) [reply src/frontend/src/styles/main.css:261]: Good point, done!

File: src/frontend/src/styles/main.css
Comment (lmuntaner) [reply src/frontend/src/styles/main.css:323]: Good idea, done!

File: src/frontend/src/components/landingPage.ts
Comment (lmuntaner) [reply src/frontend/src/components/landingPage.ts:38]: what do you mean?

File: src/frontend/src/components/landingPage.ts
Comment (lmuntaner) [reply src/frontend/src/components/landingPage.ts:28]: either what?

I added an `aria-label` makes sense to me.

**Comment by lmuntaner** [comment]
@meodai ready for another review. Thanks!

File: src/frontend/src/components/landingPage.ts
Comment (meodai) [reply src/frontend/src/components/landingPage.ts:28]: either a title (h1, h2, h3...) or aria-labael. sorry

File: src/frontend/src/components/landingPage.ts
Comment (meodai) [reply src/frontend/src/components/landingPage.ts:38]: main is only supposed to be the actual main content of the page. Not everything including the logo and this marketing stuff on the left

File: src/frontend/src/styles/main.css
Comment (meodai) [reply src/frontend/src/styles/main.css]: But then something else is the issue. i you use height and the content happens to be higher as the actual layout, this might lead to issues

**Review by meodai — APPROVED** [review]

File: src/frontend/src/components/landingPage.ts
Comment (lmuntaner) [reply src/frontend/src/components/landingPage.ts:28]: Ahh, ok. In that case, yes, the section will have a title already. The component inside always has an `h1`.

File: src/frontend/src/styles/main.css
Comment (lmuntaner) [reply src/frontend/src/styles/main.css]: The UI is ok, it's the scroll that seems weird.

This is with `min-height` (notice the pause in the scroll):

https://github.com/dfinity/internet-identity/assets/4550653/b70f5023-ab15-4f43-a1c8-6569f3e50537

This is with `height`:

https://github.com/dfinity/internet-identity/assets/4550653/8e284eee-0104-452f-bbd7-dfcdfb8a8c8a

**Pull request merged**
